processor/processor_test.go

```diff
@@ -0,0 +1,241 @@
+package processor
```

Comment:
Thanks for adding all these tests! 👍 

There's no test for `GetRoleMapping`, maybe not strictly necessary though as you're testing the other methods but if not too painful could be nice.